haystack/nodes/prompt/prompt_node.py

```diff
@@ -509,19 +509,19 @@ def prompt(self, prompt_template: Optional[Union[str, PromptTemplate]], *args, *
             for prompt in template_to_fill.fill(*args, **kwargs):
                 kwargs_copy = copy.copy(kwargs)
                 # and pass the prepared prompt and kwargs copy to the model
-                prompt = self.prompt_model._ensure_token_limit(prompt)
```

Comment:
Why rename this var? If there is no particular reason but your preference, we can revert to minimize the diff change :-)

Reply:
mypy complains about the two assignments: `prompt = self.prompt_model._ensure_token_limit(prompt)`

`Incompatible types in assignment (expression has type "Union[str, List[Dict[str, str]]]", variable has type "str")  [assignment]`

I'll try to find a better solution...